platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -110,9 +151,84 @@ def get_firmware_version(self):
         return result
 
 
+    def _check_onie_version(self):
+        # check ONIE version. To update ONIE requires version 5.2.0016 or later.
+        machine_conf = self._read_generic_file(MACHINE_CONF_FILE, MACHINE_CONF_MAX_SIZE)
```

Comment:
@stephenxs why do you need this? You already have machine_info = get_machine_info()